doc/developer-guide/documentation/RPC.en.rst

```diff
@@ -0,0 +1,397 @@
+
+=======================
+Serialization Mechanism
+=======================
+
+.. class:: MgmtMarshall
+
+   This is the class used to marshall data objects. It provides functions to marshall and unmarshall data. Each data object is associated with a field. Fields are ``MgmtMarshallType``:
+
+.. type:: MgmtMarshallType
+
+.. c:macro:: MGMT_MARSHALL_INT
+.. c:macro:: MGMT_MARSHALL_LONG
+.. c:macro:: MGMT_MARSHALL_STRING
+.. c:macro:: MGMT_MARSHALL_DATA
+
+Serialized data is sent over sockets. This is the general serialization mechanism for RPC communcation. 
+
+Generally, for remote clients sending messages to :program:`traffic_server`, the message is serialized using remote RPC APIs. The serialized message is send to :program:`traffic_manager` and :program:`traffic_manager` then simply relays the message to :program:`traffic_server`. :program:`traffic_server` eventually unserializes the message. 
+
+Details for how :program:`traffic_manager` and :program:`traffic_server` communicate are documented in the next section.
+
+Marshalling:
+============
+
+.. function:: ssize_t MgmtMarshall::mgmt_message_marshall(void *buf, size_t remain, const MgmtMarshallType *fields, unsigned count, ...)
+
+   Variable argument wrapper for ``mgmt_message_marshall_v``. Allows for different datatypes to be marshalled together as long as a field is specified for each data object. Arguments should be references to objects to be marshalled. 
+
+.. function:: ssize_t MgmtMarshall::mgmt_message_marshall_v(void *buf, size_t remain, const MgmtMarshallType *fields, unsigned count, va_list ap)
+
+   This function goes through all the data objects and serializes them together into a buffer. Based on the field, the number of bytes is determined and if there is enough space, it is written into the buffer. 
+
+    * MGMT_MARSHALL_INT: 4 bytes.
+    * MGMT_MARSHALL_LONG: 8 bytes.
+    * MGMT_MARSHALL_STRING: 4 bytes to indicate the string size in bytes, followed by the entire string and NULL terminator. 
+    * MGMT_MARSHALL_DATA: 4 bytes to indicate data size in bytes, followed by the entire data object. 
+
+Unmarshalling:
+==============
+
+.. function:: ssize_t MgmtMarshall::mgmt_message_parse(const void *buf, size_t len, const MgmtMarshallType *fields, unsigned count, ...)
+
+   Variable argument wrapper for ``mgmt_message_parse_v``. Reference to data object to store unmarshalled message needed for variable arguements. 
+
+.. function:: ssize_t MgmtMarshall::mgmt_message_parse_v(const void *buf, size_t len, const MgmtMarshallType *fields, unsigned count, va_list ap)
+   
+   This function parses all the serialized. Based on the field, the number of bytes to be read is determined and copied into a ``MgmtMarshallAnyPtr``.
+
+    * MGMT_MARSHALL_INT: 4 bytes.
```

Comment:
Are these different from the ones in the previous marshaling section? If not, it should be broken out elsewhere in a more general discussion of serialized data formatting.

Reply:
It specifies the composition of each `MgmtMarshallType`. I had it for marshalling and unmarshalling but makes more sense to just describe the macro.